File: 1544.cpp
struct Solution {
    string makeGood(string &s) {
        string res;
        for (char &c : s){
            if (res.size() == 0 || abs(c - res.back()) != 32){
                res.push_back(c);
            } else if (res.size() != 0){
                res.pop_back();
            }
        }
        return res;
    }
};

File: sol/1544/1544.py
class Solution:
    def makeGood(self, s: str) -> str:
        stack = []
        for c in s:
            if len(stack) == 0 or abs(ord(c) - ord(stack[-1])) != 32:
                stack.append(c)
            elif len(stack) != 0:
                stack.pop()
        return ''.join(stack)
